agent/entrypoint/src/main/java/co/elastic/otel/agent/ElasticAgent.java

```diff
@@ -24,12 +24,34 @@
 /** Elastic agent entry point, delegates to OpenTelemetry agent */
 public class ElasticAgent {
 
-  @SuppressWarnings("unused")
+  /**
+   * Entry point for -javaagent JVM argument attach
+   *
+   * @param agentArgs agent arguments
+   * @param inst instrumentation
+   */
   public static void premain(String agentArgs, Instrumentation inst) {
     OpenTelemetryAgent.premain(agentArgs, inst);
   }
 
+  /**
+   * Entry point for runtime attach
+   *
+   * @param agentArgs agent arguments
+   * @param inst instrumentation
+   */
+  public static void agentmain(String agentArgs, Instrumentation inst) {
+    OpenTelemetryAgent.agentmain(agentArgs, inst);
+  }
```

Comment:
[for reviewer] there is a different entry point when doing runtime attach, which is not needed otherwise.